app/server/beam/tau/lib/tau/application.ex

```diff
@@ -4,10 +4,60 @@ defmodule Tau.Application do
   @moduledoc false
 
   use Application
+  require Logger
 
   @impl true
   def start(_type, _args) do
+    Logger.info("All systems booting....")
+
+    _tau_env                       = extract_env("TAU_ENV",                            :string, "prod")
+    midi_enabled                   = extract_env("TAU_MIDI_ENABLED",                   :bool, false)
+    link_enabled                   = extract_env("TAU_LINK_ENABLED",                   :bool, false)
+    cues_on                        = extract_env("TAU_CUES_ON",                        :bool, true)
+    osc_in_udp_loopback_restricted = extract_env("TAU_OSC_IN_UDP_LOOPBACK_RESTRICTED", :bool, true)
+    midi_on                        = extract_env("TAU_MIDI_ON",                        :bool, false)
+    link_on                        = extract_env("TAU_LINK_ON",                        :bool, false)
+    osc_in_udp_port                = extract_env("TAU_OSC_IN_UDP_PORT",                :int,  5000)
+    api_port                       = extract_env("TAU_API_PORT",                       :int,  5001)
+    spider_port                    = extract_env("TAU_SPIDER_PORT",                    :int,  5002)
+    daemon_port                    = extract_env("TAU_DAEMON_PORT",                    :int,  -1)
+    daemon_token                   = extract_env("TAU_DAEMON_TOKEN",                   :int,  -1)
+    daemon_host = {127,0,0,1}
+
+    if midi_enabled do
+      Logger.info("Initialising MIDI native interface")
+      :sp_midi.init()
+    else
+      Logger.info("Starting without MIDI native interface")
+    end
+
+    if link_enabled do
+      Logger.info("Initialising Link native interface")
+      :sp_link.init()
+    else
+      Logger.info("Starting without Link native interface")
+    end
+
+    :tau_server_sup.set_application_env(
+      midi_enabled,
+      link_enabled,
+      cues_on,
+      osc_in_udp_loopback_restricted,
+      midi_on,
+      link_on,
+      osc_in_udp_port,
+      api_port,
+      spider_port,
+      daemon_port,
+      daemon_token,
+      daemon_host
+    )
+
+    Logger.info("Starting Phoenix server")
+
     children = [
+      # Start the supervision tree from Erlang modules
+      :tau_server_sup,
```

Comment:
So the secret to mounting any module in Elixir's supervision tree is to define a `child_spec(_)` function, which we do later on.